File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def check(node):
            if not node:
                return (0, True)
            left, lres = check(node.left)
            right, rres = check(node.right)
            if not lres or not rres:
                return (0, False)
            if abs(left - right) > 1:
                return (0, False)
            return (1 + max(left, right), True)
        _, res = check(root)
        return res

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    const check = (node) => {
        if (!node) return [0, true];
        const [left, lres] = check(node.left);
        const [right, rres] = check(node.right);
        if (!lres || !rres) return [0, false];
        if (Math.abs(left - right) > 1) return [0, false];
        return [1 + Math.max(left, right), true];
    }
    const [_, res] = check(root);
    return res;
};